api/users/serializers.py

```diff
@@ -102,6 +102,11 @@ def update(self, instance, validated_data):
                 setattr(instance, attr, value)
         try:
             instance.save()
+        # When a ValidationError is raised for 'social', e.args[0] will a
+        # dict of the form {'social': [<Exception>]}
+        except ValidationError as e:
+            if 'social' in e.args[0]:
+                raise InvalidModelValueError(detail=e.args[0]['social'][0][0])
```

Comment:
Ugh, I didn't catch the subclass issue. I've flipped the if so that subclass is caught first.

As to swallowing ```ValidationError``` s, I had done that on purpose. I only wanted to catch the error I fully understood. In any case I've now added more elaborate logic that should catch any other ```ValidationError``` s who's args take the same form as the social error. Anything that doesn't match gets re-raised as is.

  This will likely have the effect of [many current errors](https://sentry.cos.io/sentry/osf/?query=is%3Aunresolved+ValidationError) returning 400 with an error message instead of 500.